core/src/main/java/io/github/tcdl/msb/collector/Collector.java

```diff
@@ -253,6 +238,30 @@ private int setResponsesRemainingForResponderId(String responderId, Integer resp
         return responsesRemainingById.get(responderId);
     }
 
+    private int getWaitForResponsesFromConfigs(RequestOptions requestOptions) {
+        if (requestOptions.getWaitForResponses() == null || requestOptions.getWaitForResponses() == WAIT_FOR_RESPONSES_UNTIL_TIMEOUT) {
+            // use for Infinity number or expected responses
+            return Integer.MAX_VALUE;
```

Comment:
Still I don't understand. What is going to happen if you set waitForRespones -1, but you get back 3 billion respones? Will you be able to process the all?